test/e2e/upgrade/adminack/adminack.go

```diff
@@ -0,0 +1,134 @@
+package adminack
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	exutil "github.com/openshift/origin/test/extended/util"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+)
+
+// UpgradeTest contains artifacts used during test
+type UpgradeTest struct {
+	oc     *exutil.CLI
+	config *restclient.Config
+}
+
+func (UpgradeTest) Name() string { return "check-for-admin-acks" }
+func (UpgradeTest) DisplayName() string {
+	return "[bz-Cluster Version Operator] Verify presence of admin ack gate blocks upgrade until acknowledged"
+}
+
+// Setup creates artifacts to be used by Test
+func (t *UpgradeTest) Setup(f *framework.Framework) {
+	g.By("Setting up admin ack test")
+	oc := exutil.NewCLIWithFramework(f)
+	t.oc = oc
+	config, err := framework.LoadConfig()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	t.config = config
+	framework.Logf("Admin ack test setup complete")
+}
+
+// Test simply returns successfully if admin ack functionality is not part the baseline being tested. Otherwise,
+// test first verifies that Upgradeable condition is false for correct reason and with correct message. It then
+// modifies the admin-acks configmap to ack the necessary admin-ack gate and then waits for the Upgradeable
+// condition to change to true.
+func (t *UpgradeTest) Test(f *framework.Framework, done <-chan struct{}, upgrade upgrades.UpgradeType) {
+	// Check if this release has admin ack functionality.
+	gateCm, err := t.oc.AdminKubeClient().CoreV1().ConfigMaps("openshift-config-managed").Get(context.TODO(), "admin-gates", metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			framework.Failf(fmt.Sprintf("Error accessing configmap openshift-config-managed/admin-gates, err=%v", err))
+		} else {
+			framework.Logf("Skipping admin ack test. Admin ack is not in this baseline.")
+			return
+		}
+	}
+	var msg string
+	if msg = gateCm.Data["ack-4.8-kube-1.22-api-removals-in-4.9"]; msg == "" {
+		framework.Failf("Configmap openshift-config-managed/admin-gates does not contain ack-4.8-kube-1.22-api-removals-in-4.9 gate")
+	}
+	c, err := configv1client.NewForConfig(t.config)
+	if err != nil {
+		framework.Failf(fmt.Sprintf("NewForConfig client request failed, err=%v", err))
+	}
+	cv, err := c.ConfigV1().ClusterVersions().Get(context.TODO(), "version", metav1.GetOptions{})
+	if err != nil {
+		framework.Failf(fmt.Sprintf("Error accessing clusterversion, err=%v", err))
+	}
+	// Verify Upgradeable is false and set correctly
+	cond := getUpgradeableStatusCondition(cv.Status.Conditions)
+	checkUpgradeableStatusCondition(cond, msg)
```

Comment:
openshift/release#22368 will break your assumption that we are going into the update test with the ack unset.  Perhaps the test can be reshuffled like:

1. Check the current ack value, and confirm that the cluster is behaving as expected for the current ack setting.
2. If the ack is unset, set it, and confirm that the cluster is goes `Upgradeable=True`.

We'd exercise the vanilla 4.8 case during our 4.7 -> 4.8 CI, because for 4.7 and earlier installs, openshift/release#22368 would be a no-op.

Reply:
This still needs to get addressed, because now that openshift/release#22368 has landed, we're failing with][1]:

```
disruption_tests: [bz-Cluster Version Operator] Verify presence of admin ack gate blocks upgrade until acknowledged | 1s
-- | --
Sep 30 18:51:09.381: Upgradeable status condition not found  github.com/openshift/origin/test/e2e/upgrade/adminack.(*UpgradeTest).Test(0xc000a102f0, 0xc00115b8c0, 0xc000e5a1e0, 0x2) 	github.com/openshift/origin/test/e2e/upgrade/adminack/adminack.go:74 +0x37d 
...
```

Instead, we should be noticing that the ack is already set, and then celebrating the lack of an `Upgradeable` condition (semantically equivalent to an explicit `Upgradeable=True`).

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26446/pull-ci-openshift-origin-release-4.8-e2e-aws-upgrade/1443635824443789312